### PR TITLE
chore: migrating storybook jsx to typescript #18100

### PR DESCRIPTION
**Pull request by jayakrishnankk**
### SUMMARY
Migrated EmbedCodeButton to TypeScript to apply direction outlined in #18100 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
No changes

### TESTING INSTRUCTIONS
Storybook can be used to verify those changes.

### ADDITIONAL INFORMATION
- [x] Has associated issue: #18100
- [ ] Required feature flags: no
- [ ] Changes UI: no
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351)): no
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API: no
- [ ] Removes existing feature or API: no

tagging @ad-m , @zhaoyongjie , @kristw

### CODE REVIEW

File: superset-frontend/packages/superset-ui-demo/storybook/stories/superset-ui-number-format/BigNumberStories.tsx [hunk 1]
```diff
@@ -17,41 +17,37 @@
...
 
-/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from 'react';
 import { formatNumber } from '@superset-ui/core';
 
-const propTypes = {};
-const defaultProps = {};
-
 class NumberFormatValidator extends React.PureComponent {
+  state: { formatString: string; testValues: any[] } = {
```
Comment (zhaoyongjie) [inline superset-frontend/packages/superset-ui-demo/storybook/stories/superset-ui-number-format/BigNumberStories.tsx:24]:
```suggestion
  state: { formatString: string; testValues: (number | null | undefined)[] } = {
```

Comment (zhaoyongjie) [inline superset-frontend/packages/superset-ui-demo/storybook/stories/superset-ui-number-format/BigNumberStories.tsx:24]:
```suggestion
  state: { formatString: string; testValues: (number | null | undefined)[] } = {
```

File: superset-frontend/packages/superset-ui-demo/storybook/stories/superset-ui-time-format/TimeFormatStories.tsx [hunk 2]
```diff
@@ -17,30 +17,25 @@
...
 
-/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from 'react';
 import { formatTime } from '@superset-ui/core';
 
-const propTypes = {};
-const defaultProps = {};
-
 class TimeFormatValidator extends React.PureComponent {
+  state: { formatString: string; testValues: any[] } = {
```
Comment (zhaoyongjie) [inline superset-frontend/packages/superset-ui-demo/storybook/stories/superset-ui-time-format/TimeFormatStories.tsx:24]: same as above.
```suggestion
  state: { formatString: string; testValues: (Date | number | null | undefined)[] } = {
```

**Review by zhaoyongjie — COMMENTED** [review]
LGTM, in order to avoid `any` type, but no blocking.

File: superset-frontend/packages/superset-ui-demo/storybook/stories/superset-ui-time-format/TimeFormatStories.tsx (hunk 2 above)
Comment (jayakrishnankk) [reply superset-frontend/packages/superset-ui-demo/storybook/stories/superset-ui-time-format/TimeFormatStories.tsx:24]: This typing do suggest another change in the file
`superset-frontend/packages/superset-ui-core/src/time-format/TimeFormatterRegistrySingleton.ts::formatTime() method`

`value: Date | number | null | undefined, --> value: Date | null | undefined,` Hope it is okay to change this. Functionally it seem to work. So making that change also.

File: superset-frontend/packages/superset-ui-demo/storybook/stories/superset-ui-number-format/BigNumberStories.tsx (hunk 1 above)
Comment (jayakrishnankk) [reply superset-frontend/packages/superset-ui-demo/storybook/stories/superset-ui-number-format/BigNumberStories.tsx]: Done

**Review by zhaoyongjie — APPROVED** [review]

**Pull request merged**